### ISSUE TITLE
should adding `spring-cloud-starter-kubernetes` to pom.xml load config?

### ISSUE THREAD

**Issue by paulczar** [spring-cloud/spring-cloud-kubernetes#338]
I'm looking at spring cloud kubernetes integrations and was under the impression from the docs that all I needed was to add it to my pom.xml like so:

```
        <dependency>
            <groupId>org.springframework.cloud</groupId>
            <artifactId>spring-cloud-starter-kubernetes</artifactId>
            <version>1.0.0.RELEASE</version>
        </dependency>
```

after much frustrating debugging I discovered that what I really needed was:

```
        <dependency>
            <groupId>org.springframework.cloud</groupId>
            <artifactId>spring-cloud-starter-kubernetes-config</artifactId>
            <version>1.0.0.RELEASE</version>
        </dependency>
```

Once I added that instead of the base kubernetes it worked as expected loading values from a configmap.

Is this expected functionality? If so we should update the docs to be more specific.

**Comment by spencergibb** [comment]
Nope, if the docs say otherwise it should be changed.

**Comment by spencergibb** [comment]
Paul, do you mind pointing to the section you were reading?

**Comment by paulczar** [comment]
As you read through README.md at 
https://github.com/spring-cloud/spring-cloud-kubernetes

The first section covers [discovery](https://github.com/spring-cloud/spring-cloud-kubernetes#discoveryclient-for-kubernetes) where it says to include `<artifactId>spring-cloud-starter-kubernetes</artifactId>` in pom.xml.

The second section goes into [config settings](https://github.com/spring-cloud/spring-cloud-kubernetes#configmap-propertysource) and doesn't specify any other inclusions, so my assumption was the above inclusion was enough.

**Comment by spencergibb** [comment]
Thanks, AFK so that helps others who might fix it. I think that's the way it may have been pre release and that didn't get updated along with it.

**Comment by pukkaone** [comment]
Pull request https://github.com/spring-cloud/spring-cloud-kubernetes/pull/332 adds the `spring-cloud-starter-kubernetes-config` dependency to the documentation.

**Comment by paulczar** [comment]
wicked @pukkaone that would have solved me a lot of time and effort!

**Comment by ryanjbaxter** [comment]
Merged #332 so closing this
